### ISSUE TITLE
Assembly-Loading: Interop.HostPolicy.corehost_resolve_component_dependencies causes System.Runtime.Loader.AssemblyDependencyResolver.ctor to Fail

### ISSUE THREAD

**Issue by rakcopa** [dotnet/runtime#47799]
### Description

**Context**
I'm migrating a C++/CLR DLL from .NET Framework 4.7.2 to .NET 5. 
The DLL is a native extension that implements a certain C-interface which allows it to be used by some PowerBuilder-application.
When the application accesses functionality of the DLL it dynamically loads the DLL into the process and makes calls to the C-interface. When it is done using the extension, the DLL is unloaded. 

**Issue**
During loading of the DLL the .net runtime instantiates `System.Runtime.Loader.AssemblyDependencyResolver` which calls 
`Interop.HostPolicy.corehost_resolve_component_dependencies`, i.e. 
```csharp 
// Call hostpolicy to do the actual work of finding .deps.json, parsing it and extracting
// information from it.
returnCode = Interop.HostPolicy.corehost_resolve_component_dependencies(
    componentAssemblyPath,
    (assemblyPaths, nativeSearchPaths, resourceSearchPaths) =>
    {
        assemblyPathsList = assemblyPaths;
        nativeSearchPathsList = nativeSearchPaths;
        resourceSearchPathsList = resourceSearchPaths;
    });
```
Unfortunately the `assemblyPaths` passed by `Interop.HostPolicy.corehost_resolve_component_dependencies` to the callback delegate contains the path to the DLL twice, once with the DLLs name in all-lower-case and once with its original name. After splitting the string using `SplitPathsList(assemblyPathsLists)` the two identical paths and filenames are added to a dictionary, which of course throws a `System.ArgumentException` with the message _"An item with the same key has already been added."_

```csharp
string[] assemblyPaths = SplitPathsList(assemblyPathsList);

// Assembly simple names are case insensitive per the runtime behavior
// (see SimpleNameToFileNameMapTraits for the TPA lookup hash).
_assemblyPaths = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
foreach (string assemblyPath in assemblyPaths)
{
    _assemblyPaths.Add(Path.GetFileNameWithoutExtension(assemblyPath), assemblyPath);
}
```

The following shows an example of the content returned by `corehost_resolve_component_dependencies` whereat `demoapp.exe` is the host-application that tries to load `Extension.dll`:

> D:\Entw\Demo\Extension.dll
> D:\Entw\Demo\extension.dll
> D:\Entw\Demo\Ijwhost.dll
> D:\Entw\Demo\demoapp.exe

Luckily one can circumvent this issue by giving the extension DLL an all-lower-case name. In this case the call to 'corehost_resolve_component_dependencies' yields the path to the dll only once

> D:\Entw\Demo\extension.dll
> D:\Entw\Demo\Ijwhost.dll
> D:\Entw\Demo\demoapp.exe

and the dll can be successfully loaded and used.

Either the external function `Interop.HostPolicy.corehost_resolve_component_dependencies` should sanitize its output by removing duplicates with respect to case insensitive string comparison or `System.Runtime.Loader.AssemblyDependencyResolver` should take into account that path duplicates may appear.

### Configuration

C++/CLR Library
.NET 5.0.2 (Windows)
x86 Architecture
Microsoft Windows [Version 10.0.18363.1198]

**Comment by vitek-karas** [comment]
This is a duplicate of https://github.com/dotnet/runtime/issues/37162. Which has been fixed in https://github.com/dotnet/runtime/pull/42055 for .NET 6. For now we have no plans to backport this to 5, since a reasonable workaround exists.
